src/call.ts

```diff
@@ -118,6 +118,16 @@ export class OngoingCallPromise extends OngoingCall {
       rawResponse?: RawResponseType
     ) => {
       if (err) {
+        if (err.metadata?.internalRepr) {
+          const decoder = new GoogleErrorDecoder();
+          err.metadata.internalRepr.forEach((value: Buffer[], key: string) => {
+            if (key === 'grpc-status-details-bin') {
+              const statusDetails = decoder.decodeRpcStatusDetails(value);
+              err.statusDetails =
+                statusDetails.length === 1 ? statusDetails[0] : statusDetails;
```

Comment:
Could you help me understand what do you try to do here?
We should keep return the same type no matter how much status details here. If only one status details, we return array of status detail.